python_scripts/logging_functions.py

```diff
@@ -5,33 +5,39 @@
 import boto3
 
 s3_client = boto3.client("s3")
-
+from dataengineeringutils3.s3 import s3_path_to_bucket_key
+
+class ContextFilter(logging.Filter):
+    """
+    This is just overkill to apply a default context param to the log.
+    But it does mean I don't have to define extra everytime I wanna log.
+    So keeping it.
+    """
+    
+    def filter(self, record):
+        if not getattr(record, "context", None):
+            record.context = "PROCESSING"
+        return True
 
 def logging_setup():
-    log_name = "log.csv"
-    log_name_timestamped = f"{datetime.now().strftime('%Y-%m-%d %H:%M:%S')} log.csv"
 
-    log = logging.getLogger(log_name)
+    log = logging.getLogger("root")
     log_stringio = io.StringIO()
     handler = logging.StreamHandler(log_stringio)
-    log.addHandler(handler)
     log.setLevel(logging.INFO)
-    log.formatter = logging.Formatter(
-        fmt="%(levelname)s,%(asctime)s,%(message)s", datefmt="%Y-%m-%d %H:%M:%S"
-    )
 
-    output = {
-        "log": log,
-        "log_name_timestamped": log_name_timestamped,
-        "log_stringio": log_stringio,
-    }
-
-    return output
+    log_formatter = logging.Formatter(
+        fmt="%(asctime)s | %(module)s | %(context)s | %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
```

Comment:
should probably also keep levelname in there, to differentiate between `info`, `warning` and `error`